### PR TITLE
Add support for $g-columns as array

### PR DESCRIPTION
**Pull request by dangodev**
I’m actually surprised this works. Yay Sass!

Proposed fix for #18. And G-Unit is now part of MVCSS. Win-win.

### CODE REVIEW

**Comment by drewbarontini** [comment]
No, because if you set `$g-columns: 4`, you want this:

```css
.g-b--1of1 {
  width: 100%;
}

.g-b--1of2, .g-b--2of4 {
  width: 50%;
}

.g-b--1of3 {
  width: 33.333%;
}

.g-b--2of3 {
  width: 66.666%;
}

.g-b--1of4 {
  width: 25%;
}

.g-b--3of4 {
  width: 75%;
}
```

Your update gives you this:

```css
.g-b--1of1 {
  width: 100%;
}

.g-b--1of4 {
  width: 25%;
}

.g-b--2of4, .g-b--1of2 {
  width: 50%;
}

.g-b--3of4 {
  width: 75%;
}
```

It might be best to just go back to good old, long, explicit setup. We're always copying/pasting this in anyways.

**Comment by dangodev** [comment]
@drewbarontini You can easily add that with specifying `$g-columns: 3, 4` for specifying every grid division that you want. I wrote it this way because speaking personally, for a common design I’ll stick with the same grid system and typically everything that defies the grid has other exceptions as well that warrant it being its own structure with its own breakpoint rules (typically).

However, if you wanted to just generate _everything_ up _until_ you reach `$g-columns` (e.g., for `12`: `*of2`, `*of3`, … `*of7`, … `*of12`), then you’d do the following:

```
// _config.sass
$g-columns: 12
```

```
// _grid.sass
…

  @for $unit from 1 to $g-columns + 1
    @for $column from 1 to $unit
      #{$g-selector}g-b--#{$namespace}#{$column}of#{$unit}
        width: percentage($column / $unit)

      @for $division from 1 to $unit
        @if $column % $division == 0 and $unit % $division == 0
          #{$g-selector}g-b--#{$namespace}#{$column / $division}of#{$unit / $division}
            @extend #{$g-selector}g-b--#{$namespace}#{$column}of#{$unit}

```
This will generate styles you may not necessarily use, but is this closer to expected behavior?

**Comment by drewbarontini** [comment]
Yes, that does appear to be it! Plugging that in gave me the expected behavior.

I do understand what you're saying, but this was just a patch because I assumed it was outputting the same as the grid before it (mostly), with far more intelligence. Technically, we broke backwards compatibility, so we'll at least need to patch this, and then we can look at the grid in major/minor versions moving forward, if you have some ideas for altering its behavior.

**Comment by dangodev** [comment]
@drewbarontini Cool. Yeah, you can do either of 2 things: either change the codebase to **always** generate the full scaffold, as in the Gist, or you can change the default value for `$g-columns`:

```
$g-columns: `7, 8, 9, 10, 11, 12`
```

I personally would suggest just changing the default value, as that will give you more flexibility.

**Comment by drewbarontini** [comment]
I think we need to do the full scaffold version _for now_ to keep backwards compatibility, and then open up a `4.1` or `5.0` issue for the default value change, which would break the backwards compatibility. I do like the idea of what you're doing here, but we can't do a patch version with that change yet.

**Comment by dangodev** [comment]
:+1:

**Comment by drewbarontini** [comment]
Once you get that in here, I'll merge it in and get the version updated.

Thanks for taking a look at this!

**Comment by drewbarontini** [comment]
Actually, leave this PR open for the issue that I'll open up. Do another PR with the current fix, if you can.

**Comment by dangodev** [comment]
Done. I also pushed a fix for this PR; the `@if $column != $g-unit` line was actually unnecessary, and I changed my mind about `$g-unit`—no need to give that variable a prefix when it was only used locally, and not configurable anyway.

**Comment by drewbarontini** [comment]
:+1: Merged in, updated version, opened issue linking to this PR for 4.1 consideration.

**Comment by drewbarontini** [comment]
@dangodev Once this is ready, you can merge this branch into `v5_alpha` and close this PR since we won't want to merge it into `master`.

File: mvcss/modules/_grid.sass [hunk 1]
```diff
@@ -30,6 +30,7 @@
 
 #{$g-selector}mv-g-b
   flex: none 1
+  order: 100
```
Comment (dangodev) [inline mvcss/modules/_grid.sass]: `100` could be any number; it just makes the ordering below a little more sane

File: mvcss/modules/_grid.sass [hunk 2]
```diff
@@ -89,6 +90,31 @@
...
+    order: 107
+  #{$g-selector}mv-g-b--ninth#{$namespace}
+    order: 108
+  #{$g-selector}mv-g-b--tenth#{$namespace}
+    order: 109
+  #{$g-selector}mv-g-b--eleventh#{$namespace}
+    order: 110
+  #{$g-selector}mv-g-b--twelvth#{$namespace}
+    order: 111
+
```
Comment (dangodev) [inline mvcss/modules/_grid.sass]: @drewbarontini @nickawalsh @johndjameson What say you to naming?

Comment (johndjameson) [inline mvcss/modules/_grid.sass]: I’d favor something a little shorter: `.mv-g-b--1st`, `.mv-g-b--2nd`, etc.

**Comment by johndjameson** [comment]
Including ordering in the default grid seems a little edge case. A more built-out v5 grid would probably be better more suited to the MVCSS Library.

**Comment by johndjameson** [comment]
I merged most of the core Grid styles into `v5_alpha`, so I’m closing this PR to get things moving.

**Pull request closed without merging**